cmd/util/ledger/util/payload_file_test.go

```diff
@@ -0,0 +1,272 @@
+package util_test
+
+import (
+	"bytes"
+	"crypto/rand"
+	"path/filepath"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type keyPair struct {
+	key   ledger.Key
+	value ledger.Value
+}
+
+func TestPayloadFile(t *testing.T) {
+
+	const fileName = "root.payload"
+
+	t.Run("without filter", func(t *testing.T) {
+		unittest.RunWithTempDir(t, func(datadir string) {
+			size := 10
+
+			payloadFileName := filepath.Join(datadir, fileName)
+
+			// Generate some data
+			keysValues := make(map[string]keyPair)
+			var payloads []*ledger.Payload
+
+			for i := 0; i < size; i++ {
+				keys, values := getSampleKeyValues(i)
+
+				for j, key := range keys {
+					keysValues[key.String()] = keyPair{
+						key:   key,
+						value: values[j],
+					}
+
+					payloads = append(payloads, ledger.NewPayload(key, values[j]))
+				}
+			}
+
+			numOfPayloadWritten, err := util.CreatePayloadFile(
+				zerolog.Nop(),
+				payloadFileName,
+				payloads,
+				nil,
+			)
+			require.NoError(t, err)
+			require.Equal(t, len(payloads), numOfPayloadWritten)
+
+			payloadsFromFile, err := util.ReadPayloadFile(zerolog.Nop(), payloadFileName)
```

Comment:
I think this test case assumes the util to read the payload file is the same as the util produces it. 

This test case is necessary, however, in the real use case we will be using two utils with different versions. I think it's necessary to verify that two. 

One idea is to commit a payload file fixture produced by previous version, and add a test case to read the payload file and verify with the current version. Thoughts?

Reply:
> One idea is to commit a payload file fixture produced by previous version, and add a test case to read the payload file and verify with the current version. Thoughts?

There is only one version of payload file.  Not sure what is meant by "a payload file fixture produced by previous version".

If you are referring to payload version `payloadEncodingVersion`, we only use `payloadEncodingVersion = 1`, no other version is supported.